### PR TITLE
Use Django filesystem API

### PR DESCRIPTION
**Pull request by radimsuckr**

### CODE REVIEW

File: security/utils.py [hunk 1]
```diff
@@ -33,6 +35,8 @@
     'y': lambda amount: timezone.now() - timedelta(weeks=(52 * amount)),  # 364-day year
 }
 
+BACKUP_STORAGE = import_string(settings.BACKUP_STORAGE_CLASS)()
```
Comment (rubickcz) [inline security/utils.py]: Why is this in UPPERCASE? I don't think this is a constant, it's rather a global variable. I would write just `storage` for short.

File: security/utils.py [hunk 2]
```diff
@@ -205,18 +209,21 @@ def backup_logs_and_clean_data(self, qs, backup_path):
                 **{'{}__range'.format(self.get_timestamp_field()): (min_timestamp, max_timestamp)})
 
             if backup_path:
-                log_file_path = os.path.abspath(os.path.join(backup_path, force_text(timestamp.date())))
+                log_file_path = os.path.join(backup_path, force_text(timestamp.date()))
 
-                if os.path.isfile('{}.json.zip'.format(log_file_path)):
+                if BACKUP_STORAGE.exists('{}.json.zip'.format(log_file_path)):
```
Comment (rubickcz) [inline security/utils.py]:
```suggestion
                if storage.exists('{}.json.zip'.format(log_file_path)):
```

File: security/utils.py [hunk 3]
```diff
@@ -205,18 +209,21 @@ def backup_logs_and_clean_data(self, qs, backup_path):
...
                     i = 0
-                    while os.path.isfile('{}({}).json.zip'.format(log_file_path, i)):
+                    while BACKUP_STORAGE.exists('{}({}).json.zip'.format(log_file_path, i)):
```
Comment (rubickcz) [inline security/utils.py]:
```suggestion
                    while storage.exists('{}({}).json.zip'.format(log_file_path, i)):
```

File: security/utils.py [hunk 4]
```diff
@@ -205,18 +209,21 @@ def backup_logs_and_clean_data(self, qs, backup_path):
...
                         i += 1
                     log_file_path = '{}({})'.format(log_file_path, i)
 
                 self.stdout.write(4 * ' ' + 'generate backup file: ' + log_file_path)
-                with gzip.open('{}.json.zip'.format(log_file_path), 'wb') as file_out:
-                    file_out.write(
-                        force_bytes(json.dumps(self.serialize(qs_filtered_by_day), cls=DjangoJSONEncoder, indent=5)))
+                BACKUP_STORAGE.save(
```
Comment (rubickcz) [inline security/utils.py]:
```suggestion
                storage.save(
```

**Comment by matllubos** [comment]
@radimsuckr tohle má vést do druids/django-security, prosím pošli to tam

**Comment by matllubos** [comment]
+ rebase prosím

**Pull request closed without merging**